pkg/cmd/server/etcd/etcd.go

```diff
@@ -135,7 +135,7 @@ func TestNewEtcdClient(etcdClient newetcdclient.Client) error {
 // NeverRetryOnFailure is a retry function for the etcdClient.  If there's only one machine, master election doesn't make much sense,
 // so we don't bother to retry, we simply dump the failure and return the error directly.
 func NeverRetryOnFailure(cluster *etcdclient.Cluster, numReqs int, lastResp http.Response, err error) error {
-	if len(cluster.Machines) > 1 {
+	if len(cluster.Machines) > 1 || numReqs < 2 {
```

Comment:
Surely you find this humorous based on the name.  "How many times does this retry?", "Never, duh, look at the name", "Oh, um, maybe".


Retrying some kinds of requests leads to weird messages.  This let us see the actual errors instead of things like "object didn't exist" (when it did exist and had been removed) or "update rejected" (when it succeeded).  Why do we want to reintroduce that kind of strangeness?